topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -438,6 +437,7 @@ private void updateInternal(List<ReconcilableTopic> topics) {
 
         // skip reconciliation of paused KafkaTopics
         var partitionedByPaused = validateManagedTopics(partitionedByManaged).stream()
+            .filter(skipKafkaTopicsWithMissingSpec())
```

Comment:
Rather than a method that returns a Predicate, just make a method that _is_ the predicate and use a method reference.

But is this behaviour really right? Should we ignore these resources, or reject them with an error in the `status`

Reply:
> Rather than a method that returns a Predicate, just make a method that is the predicate and use a method reference.

Ok. Thanks.

> But is this behaviour really right? Should we ignore these resources, or reject them with an error in the status

I thought about this. My change here is to exactly match what we have in BTO. There we only raise a warning in logs, without updating the status. In a future major release, we plan to make the spec a required field by changing the schema.

I'm in favor of updating the spec, but I would prefer to do it on a dedicated PR. In the meantime, I can log it as enhancement. Wdyt?